crates/runtime/src/dataaccelerator/turso.rs

```diff
@@ -0,0 +1,2482 @@
+    ))]
+    RemoteDatabaseNotSupported,
+}
+
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+/// Connection pool for Turso databases
+#[derive(Debug)]
+pub struct TursoConnectionPool {
+    database: Arc<Database>,
+    mvcc_enabled: bool,
+    db_path: String,
+}
+
+impl TursoConnectionPool {
+    pub async fn new(path: &str, mvcc_enabled: bool) -> Result<Self> {
+        // Turso supports both file and memory modes
+        // Memory mode uses ":memory:" as the path
+        // MVCC (Multi-Version Concurrency Control) can be enabled via configuration
+        // When enabled, it supports BEGIN CONCURRENT transactions
+        let database = Builder::new_local(path)
+            .with_mvcc(mvcc_enabled)
+            .build()
+            .await
+            .context(TursoDatabaseSnafu)?;
+
+        Ok(Self {
+            database: Arc::new(database),
+            mvcc_enabled,
+            db_path: path.to_string(),
+        })
+    }
+
+    pub async fn connect(&self) -> Result<Connection> {
+        self.database.connect().context(TursoDatabaseSnafu)
+    }
+
+    /// Returns true if MVCC (Multi-Version Concurrency Control) is enabled
+    pub fn is_mvcc_enabled(&self) -> bool {
+        self.mvcc_enabled
+    }
+
+    /// Returns true if this is a memory database
+    pub fn is_memory_db(&self) -> bool {
+        self.db_path == ":memory:"
+    }
+
+    /// Returns the database path
+    pub fn db_path(&self) -> &str {
+        &self.db_path
+    }
+}
+
+/// Turso Table Provider for reading data
+#[derive(Debug)]
+pub struct TursoTableProvider {
+    schema: SchemaRef,
+    table_name: String,
+    pool: Arc<TursoConnectionPool>,
+}
+
+impl TursoTableProvider {
+    pub fn new(schema: SchemaRef, table_name: String, pool: Arc<TursoConnectionPool>) -> Self {
+        Self {
+            schema,
+            table_name,
+            pool,
+        }
+    }
+
+    /// Converts Turso database rows to Arrow RecordBatch, matching the exact schema types.
+    ///
+    /// This function is critical for reading data from Turso - it must respect the schema's
+    /// exact data types (e.g., LargeUtf8 vs Utf8, Timestamp units) to avoid type mismatches.
+    ///
+    /// Supported types:
+    /// - Integers: Int8, Int16, Int32, Int64, UInt8, UInt16, UInt32, UInt64
+    /// - Floats: Float32, Float64
+    /// - Strings: Utf8, LargeUtf8
+    /// - Binary: Binary, LargeBinary
+    /// - Boolean
+    /// - Timestamps: All time units (Second, Millisecond, Microsecond, Nanosecond)
+    /// - Dates: Date32, Date64
+    fn values_to_record_batch(
+        rows: &[Vec<TursoValue>],
+        schema: &SchemaRef,
+    ) -> Result<RecordBatch, Box<dyn std::error::Error + Send + Sync>> {
+        use arrow::array::*;
+
+        let mut columns: Vec<Arc<dyn arrow::array::Array>> = Vec::new();
```

Comment:
```suggestion
        let mut columns: Vec<Arc<dyn Array>> = Vec::new();
```

if we're importing `arrow::array::*`, why use the whole leaf here? Just use `Array`